### ISSUE TITLE
Errors when compiling scipp with Debug on Windows

### ISSUE THREAD

**Issue by rozyczko** [scipp/scipp#1244]
This is not readily reproducible - I see this on one machine (my work laptop) but my other win64 workstation compiles the code without problem (same amount of memory).
(Compilation without benchmarks and tests)

```
D:\projects\scipp\variable\inplace_arithmetic.cpp : fatal error C1128: number of sections exceeded object file
format limit: compile with /bigobj [D:\projects\scipp\build\variable\scipp-variable.vcxproj]

D:\projects\scipp\variable\arithmetic.cpp : fatal error C1128: number of sections exceeded object file format l
imit: compile with /bigobj [D:\projects\scipp\build\variable\scipp-variable.vcxproj]

cl : command line error D8040: error creating or communicating with child process [D:\projects\scipp\build\vari
able\scipp-variable.vcxproj]
```

We might want to add `/bigobj` when compiling with Debug on Windows.
The only drawback of `bigobj` is that:

> Linkers that shipped prior to Visual C++ 2005 cannot read .obj files that were produced with /bigobj.

so we are most likely safe.

**Comment by rozyczko** [comment]
When compiling with tests, I also get this error for 
`scipp-variable-test.exe`, `scipp-neutron-test.exe` and `scipp-dataset-test.exe`.
```
Error	MSB3073	The command "setlocal
"C:\Program Files\CMake\bin\cmake.exe" -D TEST_TARGET=scipp-variable-test -D TEST_EXECUTABLE=C:/Users/piotr/projects/scipp/build/variable/test/Debug/scipp-variable-test.exe -D TEST_EXECUTOR= -D TEST_WORKING_DIR=C:/Users/piotr/projects/scipp/build/variable/test -D TEST_EXTRA_ARGS= -D TEST_PROPERTIES= -D TEST_PREFIX=scipp/variable/ -D TEST_SUFFIX= -D NO_PRETTY_TYPES=FALSE -D NO_PRETTY_VALUES=FALSE -D TEST_LIST=scipp-variable-test_TESTS -D CTEST_FILE=C:/Users/piotr/projects/scipp/build/variable/test/scipp-variable-test[1]_tests.cmake -D TEST_DISCOVERY_TIMEOUT=5 -P "C:/Program Files/CMake/share/cmake-3.17/Modules/GoogleTestAddTests.cmake"
if %errorlevel% neq 0 goto :cmEnd
:cmEnd
endlocal & call :cmErrorLevel %errorlevel% & goto :cmDone
:cmErrorLevel
exit /b %1
:cmDone
if %errorlevel% neq 0 goto :VCEnd
:VCEnd" exited with code 1.	scipp-variable-test	C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets	149	
```

**Comment by rozyczko** [comment]
The issue with `setlocal` is really local file access problem and Windows permissions on target test binaries are the issue. Considering this only seems to happen on one machine, it is likely to be a fluke.

The `bigobj` issue still stands though. I need to test it further on our scipp VMs at DMSC

**Comment by Matthew-Andrew** [comment]
I can replicate the setlocal issue on my machine. Upon closer investigation however it does not appear to be a simple permissions issue. The error is caused by the post-build ctest setup as it attempts to gather the various test executables. The cause however seems to be that the test executable do not run properly but instead throw some exceptions within some of the core C++ debug dll's. 

We already set bigobj for several of the test suites see https://github.com/scipp/scipp/blob/master/variable/test/CMakeLists.txt#L42 . So I think this just needs keeping an eye on and setting in the future if it becomes necessary.

**Comment by Matthew-Andrew** [comment]
Further to this one concrete case where the setLocal error occurs is when the unit test executable fails to execute due to being unable to find the correct dll components. gtest setup attempts to run the executable to gather the test names and when this fails you end up with the slightly confusing setLocal error when the actual issue is that the test executable will not run.

**Comment by OwenArnold** [comment]
@Matthew-Andrew is this still an issue?

**Comment by Matthew-Andrew** [comment]
This has been resolved by #1641

**Comment by OwenArnold** [comment]
> This has been resolved by #1641. You still need to disable multithreading using `DDISABLE_MULTI_THREADING=On` to prevent errors due to being unable to find tbb_debug.lib but this is manageable.

I guess we can't do this at cmake time, because the build configuration is a build tool configuration in VS?
